peer.go

```diff
@@ -159,11 +159,67 @@ func (l *PeerList) Copy() map[string]*Peer {
 
 	listCopy := make(map[string]*Peer)
 	for k, v := range l.peersByHostPort {
-		listCopy[k] = v
+		listCopy[k] = v.Peer
 	}
 	return listCopy
 }
 
+// exists checks if a hostport exists in the peer list.
+func (l *PeerList) exists(hostPort string) (*peerScore, bool) {
+	l.RLock()
+	ps, ok := l.peersByHostPort[hostPort]
+	l.RUnlock()
+
+	return ps, ok
+}
+
+// UpdatePeerHeap updates the peer heap.
```

Comment:
this function is leaking implementation details of the heap to the caller. I think we should rename this to `PeerUpdated`
and the comment can be like 
```go
// PeerUpdated is called when there is a change that may cause the peer's score to change.
// The new score is calculated, and the peer heap is updated with the new score
```